learning_journal_basic1/views/default.py

```diff
@@ -0,0 +1,98 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+from sqlalchemy.exc import DBAPIError
+
+from pyramid.httpexceptions import HTTPFound
+from datetime import datetime
+
+from ..models import MyModel
+
+TIME_FORMAT = '%B %d, %Y'
+
+
+@view_config(route_name='home', renderer='../templates/list.jinja2')
+def home_view(request):
+    """Display a list of all entries in the database on Home Page."""
+    try:
+        query = request.dbsession.query(MyModel)
+        entries = query.all()
+    except DBAPIError:
+        return Response(db_err_msg, content_type='text/plain', status=500)
+    return {"entries": entries}
+
+
+@view_config(route_name='detail_view', renderer='../templates/detail.jinja2')
+def detail_view(request):
+    """Display details of a particular entry based on id."""
+    query = request.dbsession.query(MyModel)
+    entry = query.filter(MyModel.id ==
+                                    int(request.matchdict['id'])).first()
+    return {'entry': entry}
```

Comment:
What is the value of `entry` if the id your visitor requested is not in the database?  What happens to your template in that case?  What should a  proper response be if you don't have an entry that the visitor requested?